### PR TITLE
Fix stale FIXMEs for missing input/output validation

### LINKED ISSUE #217: Several nodes have stale FIXMEs for missing input/output validation
## Summary

Multiple node processors have FIXME comments for validation gaps that should be addressed. These fall into two categories:

### Missing input validation

| Node | File | Issue |
|------|------|-------|
| ConstantOfShape | `constant_of_shape.rs:112` | `value` attribute not validated to contain exactly one element |
| Trilu | `trilu.rs:10, 105` | No rank >= 2 validation on input; `k` input not validated as 0-D tensor |
| Tile | `tile.rs:14, 18` | `repeats` tensor length not validated against input rank; negative values not rejected |
| MatMulInteger | `matmulinteger.rs:49` | Optional zero-point inputs (inputs 3-4) not validated |
| Where | `where_op.rs:108` | Condition input allows Shape type (I64) when spec requires boolean |
| ConvTranspose2d | `conv_transpose2d.rs:135` | Padding order variable names may be swapped vs ONNX spec convention |

### Missing optional output support

| Node | File | Issue |
|------|------|-------|
| MaxPool1d | `max_pool1d.rs:107` | Optional `Indices` output not supported |
| MaxPool2d | `max_pool2d.rs:104` | Optional `Indices` output not supported |
| LayerNorm | `layer_norm.rs:96` | Only 1 output validated; spec allows 1-3 (Y, Mean, InvStdDev) |
| BatchNorm | `batch_norm.rs:103` | `training_mode` attribute not validated |

These are all documented with FIXME comments in the source. Each is a minor spec deviation that could cause silent failures with certain ONNX models.

### PR DESCRIPTION
**Pull request by antimora**
## Summary

Closes #217. Addresses stale FIXME comments across 10 node processors by adding missing validation, correcting `OutputSpec`, and cleaning up resolved comments.

- **ConstantOfShape**: validate `value` attribute contains exactly one element
- **Tile**: reject negative repeats values; validate `len(repeats) == rank(input)` when statically known
- **MatMulInteger**: validate input count is 2-4 (was only checking >= 2)
- **BatchNorm**: reject `training_mode=1` since training outputs are not supported
- **MaxPool1d/2d**: `OutputSpec::Exact(1)` to `Range(1, 2)` for optional Indices output
- **LayerNorm**: `OutputSpec::Exact(1)` to `Range(1, 3)` for optional Mean/InvStdDev outputs
- **ConvTranspose2d**: fix misleading pad variable names to match ONNX ordering `[top, left, bottom, right]`
- **Trilu**: remove stale FIXMEs (rank >= 2 and scalar k validation already existed)
- **Where**: replace FIXME with explanatory comment (Shape exception is necessary because comparison ops on shapes produce I64 results)

## Test plan

- [x] `cargo test -p onnx-ir` (893 tests pass)
- [x] `cargo test -p burn-onnx` (all pass)
- [x] `cargo test -p onnx-tests` (515 integration tests pass)
- [x] New unit tests for: multi-element value rejection (ConstantOfShape), negative repeats rejection (Tile), repeats length mismatch (Tile), training_mode rejection (BatchNorm), rank < 2 rejection (Trilu)

### CODE REVIEW

File: crates/onnx-ir/src/node/max_pool2d.rs [hunk 1]
```diff
@@ -88,7 +88,7 @@ impl NodeProcessor for MaxPool2dProcessor {
             min_opset: 1,
             max_opset: None,
             inputs: InputSpec::AtLeast(1),
-            outputs: OutputSpec::Exact(1),
+            outputs: OutputSpec::Range(1, 2),
         }
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/onnx-ir/src/node/max_pool2d.rs:92]: This follows the same pattern as BatchNorm, which uses `OutputSpec::Range(1, 5)` with `infer_types` only setting `outputs[0]`. The framework handles additional outputs that aren't explicitly typed. The purpose of this change is to stop rejecting valid ONNX models that declare the optional Indices output, even if codegen doesn't use it yet. Reverting to `Exact(1)` would re-introduce the original bug from #217.

File: crates/onnx-ir/src/node/max_pool1d.rs [hunk 2]
```diff
@@ -91,7 +91,7 @@ impl NodeProcessor for MaxPool1dProcessor {
             min_opset: 1,
             max_opset: None,
             inputs: InputSpec::AtLeast(1),
-            outputs: OutputSpec::Exact(1),
+            outputs: OutputSpec::Range(1, 2),
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/onnx-ir/src/node/max_pool1d.rs:94]: Same reasoning as the MaxPool2d comment: this follows the established BatchNorm pattern (`OutputSpec::Range(1, 5)` with only `outputs[0]` typed). Reverting would reject valid ONNX models with the optional Indices output.

File: crates/onnx-ir/src/node/layer_norm.rs [hunk 3]
```diff
@@ -69,7 +69,7 @@ impl NodeProcessor for LayerNormProcessor {
             min_opset: 17,
             max_opset: None,
             inputs: InputSpec::AtLeast(2),
-            outputs: OutputSpec::Exact(1),
+            outputs: OutputSpec::Range(1, 3),
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/onnx-ir/src/node/layer_norm.rs:71]: Same pattern as BatchNorm (`OutputSpec::Range(1, 5)` with only `outputs[0]` typed) and the MaxPool changes above. The framework handles untyped optional outputs without breaking downstream inference.

File: crates/onnx-ir/src/node/tile.rs [hunk 4]
```diff
@@ -89,14 +82,41 @@ impl NodeProcessor for TileProcessor {
...
+            && let Some(tensor_data) = repeats_arg.value()
+        {
+            let i64_values: Vec<i64> =
+                tensor_data
+                    .to_i64_vec()
+                    .map_err(|_| ProcessError::TypeMismatch {
+                        expected: "Int64".to_string(),
+                        actual: format!("{:?}", tensor_data.elem_type()),
+                    })?;
+
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/onnx-ir/src/node/tile.rs:96]: The `to_i64_vec()` accepting I32 is fine here since both are valid integer types for this conversion. The concern about `extract_config` using `to_vec().unwrap()` is valid but is pre-existing code, not introduced by this PR. The ONNX spec requires repeats to be int64, so the existing `extract_config` path works for spec-compliant models.

File: crates/onnx-ir/src/node/tile.rs [hunk 5]
```diff
@@ -89,14 +82,41 @@ impl NodeProcessor for TileProcessor {
...
-        // TODO: Missing validation that repeats input exists (required in opset 6+).
-        // Opset 6+ requires repeats as second input but this isn't strictly validated.
-
-        // TODO: Missing validation that len(repeats) == rank(input).
-        // ONNX spec requires repeats length to match input rank but not validated here.
+        // Validate repeats when statically known
+        if let Some(repeats_arg) = node.inputs.get(1)
+            && let Some(tensor_data) = repeats_arg.value()
+        {
+            let i64_values: Vec<i64> =
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/onnx-ir/src/node/tile.rs:89]: Valid observation, but this is a pre-existing issue with the Tile processor (the `InputSpec::AtLeast(1)` and opset-dependent input requirements existed before this PR). This PR adds validation for the repeats values when they are present and static. Tightening the InputSpec for opset >= 6 would be a good follow-up but is out of scope for this FIXME cleanup.

File: crates/onnx-ir/src/node/batch_norm.rs [hunk 6]
```diff
@@ -99,14 +99,14 @@ impl NodeProcessor for BatchNormProcessor {
...
-        // TODO: Add test for wrong input tensor ranks - spec requires scale/bias/mean/var to be 1D
-        // TODO: Validate input[3] and input[4] are actually mean and variance tensors (rank 1)
+        // Reject training mode since training-specific outputs are not supported
+        if let Some(training_mode) = node.attrs.get("training_mode")
+            && training_mode.clone().into_i64() != 0
+        {
+            return Err(ProcessError::Custom(
+                "BatchNorm: training_mode=1 is not supported (only inference mode)".to_string(),
+            ));
+        }
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/onnx-ir/src/node/batch_norm.rs:109]: The `.clone().into_i64()` pattern is used throughout the codebase for attribute extraction (e.g., trilu.rs, max_pool1d.rs ceil_mode, conv_transpose2d.rs group). The ONNX spec defines `training_mode` as an int attribute, so it will always be Int64 from valid protobuf parsing. Adding defensive handling here would be inconsistent with how every other int attribute is accessed in this project.

File: crates/onnx-ir/src/node/constant_of_shape.rs [hunk 7]
```diff
@@ -117,15 +117,17 @@ impl NodeProcessor for ConstantOfShapeProcessor {
...
+        // Validate that the value attribute contains exactly one element (per ONNX spec)
+        if let Some(value_attr) = node.attrs.get("value") {
+            let tensor = value_attr.clone().into_tensor();
+            let num_elements: usize = tensor.shape.iter().product();
+            if num_elements != 1 {
+                return Err(ProcessError::Custom(format!(
+                    "ConstantOfShape: 'value' attribute must contain exactly one element, got {}",
+                    num_elements
+                )));
+            }
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/onnx-ir/src/node/constant_of_shape.rs:129]: The `value` attribute for ConstantOfShape is a one-element tensor per spec, so its shape is `[]` (scalar) or `[1]`. A shape that overflows `usize` via `.product()` would require dimensions like `[usize::MAX, 2]`, which can't come from valid ONNX protobuf parsing. Adding checked arithmetic here would be defensive against a scenario that can't happen in practice.
